### PR TITLE
[Fix] Add sorting feature and fix filter for testcase.

### PR DESCRIPTION
**Pull request by cservakt**
The sorting feature was not available for the 'testcase' column since it first appeared in CodeChecker 6.22.2. The new PR gives a support to sort the reports on the Testcase column.

The testcase titles were truncated at the end in the filtering configuration modal if they were too long. In most cases, the unique part of the title is located at the end. The new modification allows the last part to remain intact while cutting characters at the beginning, similar to the filepath filter.

The testcase section of the report filter has listed all values from the report annotation table. With the new PR, those names appear only that belong to the given run and satisfy the filter conditions. It is also possible to search precisely for a testcase, similar to the filepath filter.

### CODE REVIEW

File: web/api/report_server.thrift [hunk 1]
```diff
@@ -593,7 +594,9 @@ service codeCheckerDBAccess {
   // Get report annotation values belonging to the given key.
   // The "key" parameter is optional. If not given then the list of keys returns.
   // PERMISSION: PRODUCT_VIEW
-  list<string> getReportAnnotations(1: optional string key),
+  list<string> getReportAnnotations(1: list<i64>    runIds,
+                                    2: ReportFilter reportFilter,
+                                    3: CompareData  cmpData),
```
[the remark this answers is not in the document]
Comment (cservakt) [reply web/api/report_server.thrift:599]: Which part of it? Would cmpData be better as an optional parameter?

[the remark this answers is not in the document]
Comment (cservakt) [reply web/api/report_server.thrift:599]: Now, I understand the problem clearly. Thanks for the suggestion.

**Comment by Szelethus** [comment]
Otherwise, this patch LGTM, but I'm not terribly confident in all libraries the patch touches.

**Review by Szelethus — APPROVED** [review]
LGTM

**Pull request merged**